
REF=RarData_xb/04.ref/csiA.fasta

### clean the data
while read i; do fastp -i $i'_1.fq.gz' -I $i'_2.fq.gz' -o $i'.cleaned_1.fq.gz' -O $i'.cleaned_2.fq.gz' -l 36 -q 20 -w 12 ; done < list

## mapping reads
bwa index $REF
while read i ; do bwa mem -t 10 -R "@RG\tID:XB\tSM:$i\tLB:WGS\tPL:Illumia" $REF \
$i'_1.cleaned.fq.gz' $i'_2.cleaned.fq.gz' | singularity exec trinityrnaseq.v2.14.0.simg samtools sort -@ 10 -o $i'.bam' - ; done < list

## MarkDuplicates
gatk --java-options "-XX:ParallelGCThreads=10" MarkDuplicates -I T1.sort.bam -M T1.markdup_metrics.txt -O T1.markdup.bam > log2dedup.T1 2>&1

## call SNP
gatk --java-options "-Xmx800m -Djava.io.tmpdir=./tmp" HaplotypeCaller  \
-R $REF -I T1.markdup.bam -O T1.g.vcf.gz -ERC GVCF

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Calculate the degree of heterozygosity in different genomic region in each individual
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

### The analysis process is the same for each individual. The following takes [calculate the heterozygosity of each exon region in individual 1] as an exampleï¼š

# call vcf and filter the SNP
gatk --java-options "-Xmx10g"   GenotypeGVCFs -R $REF --variant T1.g.vcf.gz -O T1.vcf.gz
gatk SelectVariants -select-type SNP  --restrict-alleles-to BIALLELIC -V T1.vcf.gz -O T1.snp.vcf.gz \
gatk VariantFiltration \
    -V T1.snp.vcf.gz \
    -filter "QD < 2.0" --filter-name "QD2"    -filter "QUAL < 30.0" --filter-name "QUAL30" \
    -filter "SOR > 3.0" --filter-name "SOR3"  -filter "FS > 60.0" --filter-name "FS60" \
    -filter "MQ < 40.0" --filter-name "MQ40"  -filter "MQRankSum < -12.5" --filter-name "MQRankSum-12.5" \
    -filter "ReadPosRankSum < -8.0" --filter-name "ReadPosRankSum-8" \
    -O T1.filt.snp.vcf.gz

gatk SelectVariants -R $REF -V T1.filt.snp.vcf.gz --exclude-filtered -O T1.filtered.snp.vcf.gz

### pick up the vcf in each region
awk '($3=="exon") {OFS="\t"; print $1,$4-1,$5}' CskiameaA.gff > CskiameaA_gff_exon.bed
awk '($3=="gene") {OFS="\t"; print $1,$4-1,$5,$9}' CskiameaA.gff | sed 's/ID=//g' > CskiameaA_gff_gene.bed
# awk '{if($3=="gene") {if($7=="+") {start=$4-2000; end=$4;} else {if($7=="-") start=$5; end=$5+2000; } if(start<0) start=0; print $1,start,end,$14,$10,$7,$9;}}' CskiameaA.gff | sed 's/ /\t/g' | sed 's/ID=//g' | awk '{print $1"\t"$2"\t"$3"\t"$5}'> CskiameaA_gff_promoter.bed
# awk '($3=="intron") {OFS="\t"; print $1,$4-1,$5}' /public1/node3_liushk/braker_result/03.braker_result/CskiameaA.longest.gff > CskiameaA_gff_intron.bed

vcftools --gzvcf T1.filtered.snp.vcf.gz --bed CskiameaA_gff_exon.bed --recode --out T1.filtered.snp.exon &
bcftools view -O z T1.filtered.snp.exon.recode.vcf > T1.filtered.snp.exon.vcf.gz
# bcftools index -t T1.filtered.snp.exon.vcf.gz

## split the bed file and pick the vcf in each region 
split -l 1 CskiameaA_gff_gene.bed -a 5 00.split.bed/gene_
**repl.sh in the script**
while read i ; do bash ../repl.sh $i ; done < 00.list

while read i ; do bcftools view -R '00.split.bed/'$i'.bed' T1.filtered.snp.exo.nvcf.gz > '02.exon2t1.vcf/'$i'.vcf' ; done < genelist
while read i ; do plink --vcf '02.exon2t1.vcf/'$i'.vcf' --make-bed --out '03.ready2exon2t1/'$i --allow-extra-chr ; done < genelist
ls *bim | grep -v temporary | cut -f 1 -d "." > 00.list
while read i ; do awk '{print $1"\tM"NR"\t"$3"\t"$4"\t"$5"\t"$6}' $i'.bim' > tmp && mv tmp $i'.bim' ; done < 00.list

while read i ; do ~/shicy/hiblup --bfile '03.ready2exon2t1/'$i --hete --out '04.hiblup2exon2t1/'$i --thread 2 ; done < 03.ready2exon2t1/00.list

### count the het for each gene
while read i ; do cut -f 2 '04.hiblup2exon2t3/'$i'.hete' | xargs |  cut -f 2 -d " " | awk '{print "'$i'"" " $0 }' >> 05.hete2exon2t1 ; done <  03.ready2exon2t3/00.list


####### For diff counts
### Combine the gvcf result for diversity anlysis
gatk  --java-options "-Xmx80g"   CombineGVCFs  -R $REF -O all.merge.g.vcf.gz  -V T1.merge.vcf.gz -V T2.merge.vcf.gz -V T3.merge.vcf.gz
gatk  --java-options "-Xmx20g"   GenotypeGVCFs -R $REF --variant all.merge.g.vcf.gz -all-sites -O all.merge.all-sites.vcf.gz

gatk SelectVariants -select-type SNP  --restrict-alleles-to BIALLELIC -V 08.vcf/all.merge.all-sites.vcf.gz -O 09.filted.snp/01.all.snp.vcf.gz
gatk VariantFiltration -V 01.all.snp.vcf.gz -filter "QD < 2.0" --filter-name "QD2" -filter "QUAL < 30.0" \
     --filter-name "QUAL30"  -filter "SOR > 3.0" --filter-name "SOR3" -filter "FS > 60.0" --filter-name "FS60" -filter "MQ < 40.0" --filter-name "MQ40" \
    -filter "MQRankSum < -12.5" --filter-name "MQRankSum-12.5"   -filter "ReadPosRankSum < -8.0" --filter-name "ReadPosRankSum-8"  -O 02.filt.snp.vcf.gz

vcftools --gzvcf 02.filt.snp.vcf.gz --max-maf 0 --recode --stdout --max-missing 0.6 --min-meanDP 10 --max-meanDP 500 --remove-indels  | bgzip -c > 03.test_invariant.vcf.gz
vcftools --gzvcf 02.filt.snp.vcf.gz --mac 1 --recode --stdout --max-missing 0.6 --min-meanDP 10 --max-meanDP 500 --remove-indels  | bgzip -c > 03.test_variant.vcf.gz

tabix 03.test_invariant.vcf.gz
tabix 03.test_variant.vcf.gz

gatk SelectVariants -R ../04.ref/csi.main.fasta -V 03.test_variant.vcf.gz --exclude-filtered -O 03.test_variant.filtered.vcf.gz
bcftools concat --allow-overlaps 03.test_invariant.vcf.gz 03.test_variant.filtered.vcf.gz -O z -o 04.filtered.vcf.gz

pixy --stats pi --vcf 04.filtered.vcf.gz --populations pop_ndm_all --bed_file CskiameaA_gff_gene.bed --n_cores 12 --bypass_invariant_check yes

### script
# R
compare Rscript
