# The scripts of the article: The origins and functional significance of bivalve genome diversity
These scripts were used to generate the analyses.
Details of the methods are in the paper.
Contact: C. Shi <shichenyu@stu.ouc.edu.cn> and S. Liu <liushk@ouc.edu.cn>
Note: these scripts are internal to this project, and not currently designed for generic use, so results are not guaranteed for adapt in other work

## Experimental Design
1 Individual genome assembly with PacBio HiFi data and Hi-C data

2 Population analysis with DNAseq data from C. sikamea, C. ariakensis, C. angulata, and C. gigas.

3 Allele expression analysis with four RNA-seq datasets, which include:
 a) adductor muscles, mantle, digestive gland, gill, and labial palps, were collected from three wild individuals
 b) adductor muscle, digestive gland, and labial palps were collected from four individuals in March
 c) adductor muscle, digestive gland, and labial palps were collected from three individuals in June
 d) adductor muscle, digestive gland, and labial palps were collected from five individuals in October

4 The publicly data from the NCBI and Darwin Tree of Life Project (DToL, https: //www.darwintreeoflife.org/).

## 01.genome_compare
*Comparative analysis of the genome size, gene composition, and individual genomic variation across different bivalve genomes*

## 02.GenomeAssemblyandAnnotation
*Scripts to genome Assembly and Annotation*

## 02. population_anlysis
*Scripts to analyze genetic diversity and gene flow*

## 03.four_haplotype_genome_compare
*Scripts to comparative analysis of high divergence sequences among the haplotype-resolved genomes of bivalves*

## 04.GeneCompare
*Scripts to comparative analysis of haplotype divergence function and its impact of on gene evolution and allelic expression*

## 05.alleleexpression
*Scripts to ASE expression*
