### genome assembly
nohup ~/TY/softwares/hifiasm-master-0.14.2/hifiasm -o hifiasm_sikamea_0142 -t 40 ../../hifi_m64165_201125_063852_subreads.fasta.gz 1>hifi_4_2.log 2>&1 &
awk '{if($1=="S")print ">"$2"\n"$3}' hifiasm_sikamea_0142.p_utg.gfa > hifiasm_sikamea_0142.p_utg.fasta
### Busco
singularity exec -B /public1:/public1 /public1/db/sif/busco_v5.4.7.sif busco -m genome -i genome.fa -o genome -l /public1/node3_shicy/csi_tmp/00.gigas.genome/busco_downloads/metazoa_odb10/ --offline -c 4

##  juicer + 3d-dna
seqkit fx2tab -l -g -n -i -H hifiasm_sikamea_0142.p_utg.fasta | awk '{print $1"\t"$2}' > genome.contig.size
nohup singularity exec -B /public1:/public1  ~/shicy/juicer_latest.sif juicer.sh -t 40 -z purged.fa -y genome_MboI.txt -p genome.contig.size -s MboI -D /opt/juicer &
## nohup singularity exec -B /public1:/public1  ~/shicy/juicer_latest.sif juicer.sh -t 40 -z purged.fa -y genome_HindIII.txt -p genome.contig.size -s HindIII -D /opt/juicer &
nohup bash /public/home/shicy/3d-dna-master/run-asm-pipeline.sh  -r 2 --build-gapped-map --editor-repeat-coverage 5 purged.fa merged_nodups.txt &
nohup bash ~/shicy/3d-dna/run-asm-pipeline.sh -r 2 --build-gapped-map --editor-repeat-coverage 5 purged.fa merged_nodups.txt &

bash ~/shicy/3d-dna/run-asm-pipeline-post-review.sh -r purged.FINAL.review.assembly  ../05.3d-dna/purged.fa ../05.3d-dna/merged_nodups.txt
bash ~/mambaforge/envs/hic/share/3d-dna/run-asm-pipeline-post-review.sh -r purged.rawchrom.review.assembly ../05.3d-dna/purged.fa ../05.3d-dna/purged.mnd.txt


################## genome assembly evaluation
fastp -i FDSW202364006-1r_L1_1_clean.rd.fq.gz -o csi_survey_1.cleaned.fq.gz -I FDSW202364006-1r_L1_2_clean.rd.fq.gz  -O csi_survey_2.cleaned.fq.gz -l 36 -q 20  -w 12
bwa mem -t 40 -R "@RG\tID:ND\tSM:Survey\tLB:WGS\tPL:Illumia" hifiasm_sikamea_0142.p_utg.FINAL.fasta csi_survey_1.cleaned.fq.gz csi_survey_2.cleaned.fq.gz | singularity exec -B /public1:/public1 /public1/db/sif/trinityrnaseq.v2.14.0.simg samtools sort -@ 40 -o csi_survey.sorted.bam
BamDeal_Linux statistics Coverage -i nd10.sorted.bam -r /public1/node3_shicy/PRJ02_Csikamea/00.data/00.Annotion/00.annotion/hifiasm_sikamea_0142.p_utg.FINAL.fasta -s -o depthpra_s_nd10 -w 100000

### Hic
################## HIC results evaluation
bwa mem -t 30 -A1 -B4  -E50 -L0  index/hyotis_chr.fasta merge_hic_1.fq.gz 2>> hyotis_R1.log | samtools view -@ 30 -Shb - > mate_R1.bam
bwa mem -t 30 -A1 -B4  -E50 -L0  index/hyotis_chr.fasta merge_hic_2.fq.gz 2>> hyotis_R2.log | samtools view -@ 30 -Shb - > mate_R2.bam

hicBuildMatrix --samFiles mate_R1.bam mate_R2.bam --binSize 10000 --restrictionSequence GATC --danglingSequence GATC --restrictionCutFile   DpnII.bed --threads 16 --inputBufferSize 100000 --outBam hic.bam  -o hic_matrix.h5 --QCfolder ./hicQC

hicMergeMatrixBins --matrix hic_matrix.h5 --outFileName hic_matrix.nb5.h5 -nb 5 &
hicMergeMatrixBins --matrix hic_matrix.h5 --outFileName hic_matrix.nb10.h5 -nb 10 &
hicMergeMatrixBins --matrix hic_matrix.h5 --outFileName hic_matrix.nb20.h5 -nb 20 &

hicCorrectMatrix diagnostic_plot -m hic_matrix.nb5.h5 -o hic_matrix.nb5.h5.png
hicCorrectMatrix diagnostic_plot -m hic_matrix.nb10.h5 -o hic_matrix.nb10.h5.png
hicCorrectMatrix diagnostic_plot -m hic_matrix.nb20.h5 -o hic_matrix.nb20.h5.png

hicCorrectMatrix correct -m hic_matrix.nb5.h5  --filterThreshold -2 2 -o hic_corrected.nb5.h5
hicCorrectMatrix correct -m hic_matrix.nb10.h5 --filterThreshold -2 2 -o hic_corrected.nb10.h5
hicCorrectMatrix correct -m hic_matrix.nb20.h5 --filterThreshold -2 2 -o hic_corrected.nb20.h5

## plot
hicPlotMatrix -m hic_corrected.nb10.h5 -o hicPlotMatrix.nb10.new.pdf --log1p --colorMap OrRd  --clearMaskedBins --chromosomeOrder HapA_chr5 HapB_chr5 HapA_chr3 HapB_chr3 HapA_chr2 HapB_chr2 HapA_chr6 HapB_chr6 HapA_chr7 HapB_chr7 HapA_chr9 HapB_chr9 HapA_chr4 HapB_chr4 HapA_chr1 HapB_chr1 HapA_chr10 HapB_chr10 HapA_chr8 HapB_chr8 --dpi 300
