
#### Repeatmoder
singularity exec -B /public1:/public1 /public1/db/sif/dfam-tetools-latest.sif BuildDatabase -name Cgi -engine ncbi Cgigas.fa
nohup singularity exec -B /public1:/public1 /public1/db/sif/dfam-tetools-latest.sif RepeatModeler --database Mer -threads 20  -LTRStruct 1>01.RepeatModeler.log 2>&1 &
nohup singularity exec -B /public1:/public1 /public1/db/sif/dfam-tetools-latest.sif RepeatMasker -lib Car_edta.fa.mod.EDTA.TElib.fa -no_is -norna -pa 16 -gff -xsmall genome.HK.fa &

##### BRAKER3
nohup singularity exec /home/data/braker3.sif braker.pl --genome=blk.fasta.masked --prot_seq=Metazoa.fa --rnaseq_sets_dirs=rnaseq2cgigas --softmasking --threads 32 --AUGUSTUS_CONFIG_PATH=augustus-3.4.0/config --GENEMARK_PATH=${ETP}/gmes --workingdir=braker2blk --rnaseq_sets_ids=S1,S2,S3,S4,S5,S6,S7,S8,S9 &

agat_convert_sp_gff2gtf.pl --gff tsebra1.gtf -o tsebra_result.gff
agat_sp_manage_IDs.pl --gff tsebra_result.gff --ensembl --prefix blk_ --type_dependent  -o tsebra_result.tmp.gff
awk -F ";" '!/^#/ {split($1,a,"\t");if(a[3]=="gene") print $1;else print $1";"$2 }; /^#/ {print $0}' tsebra_result.tmp.gff > tsebra_result.final.gff
agat_sp_keep_longest_isoform.pl -gff  tsebra_result.final.gff -o blk.longest.gff

agat_sp_extract_sequences.pl -g CskiameaA.longest.gff -f ../01.genome.mask/Cgigas.fa -t cds -p -o CskiameaA.pep.fa
agat_sp_extract_sequences.pl -g CskiameaA.longest.gff -f ../01.genome.mask/Cgigas.fa -t cds    -o CskiameaA.cds.fa

##### gene function annotation
emapper.py -i CskiameaB.pep.fa --output 01.CsiB -m diamond -d euk --data_dir /public1/db/eggnog-mapper-data/ --cpu 10
nohup /public1/db/interproscan/interproscan-5.60-92.0/interproscan.sh --goterms -cpu 20 -dp -i CskiameaB.pep.fa TSV,XML -d . -T CskiameaB &

############# TE annotation with EDTA
awk '{if($0~"LINE" || $0~"SINE")print$0}' ../Genome-families.fa | cut -f 1 -d " " | sed 's/>//g' > id
seqkit grep -f id ../Genome-families.fa  | awk '{print$1}'>  INE.liban

nohup EDTA.pl --genome genome.fna --cds cds.fa --species others --step all --curatedlib INE.liban --u --anno 1 --threads 10 &

cp 01.repeatmod/Genome-families.fa Genome.fa.mod.EDTA.final/Genome.fa.mod.RM.consensi.fa
nohup EDTA.pl --genome Genome.fa  --cds cds.fa --species others --step final --anno 1  --overwrite 0 --curatedlib INE.liban --sensitive 1 --threads 10 &

##### anno
EDTA.pl --genome Genome.fa  --cds cds.fa --species others --step anno --anno 1 --overwrite 1 --curatedlib INE.liban --threads 10

########### TE identify
nohup singularity exec -B /public1:/public1 /public1/db/sif/dfam-tetools-latest.sif RepeatMasker -lib Genome.fa.mod.EDTA.TElib.fa -no_is -norna -pa 20 -gff -xsmall Genome.fa &


#### For genoem annotation using galba with no RNAseq data
singularity exec ../galba.sif galba.pl --genome=Afixchr.qry.filtered.fa.masked --prot_seq=../pep.fa --workingdir=Afixchr.qry --threads 10  --AUGUSTUS_CONFIG_PATH=~/Augustus/config
